### PR TITLE
Fix workflow in integration test.

### PR DESCRIPTION
**Pull request by florian-huber**
I had missed one step in the integration test workflow. Usually I would put the ``require_minimum_number_of_peaks`` last, because other filters such as ``select_by_relative_intensity`` or ``select_by_mz`` will also potentially remove peaks (so one could end up with spectra having less peaks than required). 

When running it with the entire raw data, the adapted integration workflow failed when empty spectra were passed (not spectrums which are 'None', but spectrums that contain empty empty arrays of peaks).

- I changed the order of the filters in the integration test.
- I added tests for spectrums with empty peak arrays.
- I fixed ``normalize_intensities`` and ``select_by_relative_intensity`` so that they won't fail when facing empty intensity arrays.

**The one test that is still failing should be fixed by #155.**

### CODE REVIEW

File: tests/test_normalize_intensities.py [hunk 1]
```diff
@@ -14,3 +14,15 @@ def test_normalize_intensities():
...
     assert numpy.array_equal(spectrum.mz, mz), "Expected the spectrum's intensity locations along "\
                                                "the mz-axis to be unaffected."
+
+
+def test_normalize_intensities_empty_peaks():
+
+    mz = numpy.array([], dtype='float')
+    intensities = numpy.array([], dtype='float')
+    spectrum_in = Spectrum(mz=mz, intensities=intensities,
+                           metadata={"testfield": "TEST"})
```
Comment (jspaaks) [inline tests/test_normalize_intensities.py]: Why does this unit test have ``metadata``? The point of the test is ``mz`` and ``intensities`` being empty arrays, right?

File: tests/test_select_by_relative_intensity.py [hunk 2]
```diff
@@ -78,3 +78,16 @@ def test_select_by_relative_intensity_with_from_and_to_parameters():
...
     assert numpy.array_equal(spectrum.intensities, numpy.array([10, 100], dtype="float"))
+
+
+def test_select_by_relative_intensity_with_empty_peaks():
+    """Within certain workflows it can happen that spectrums are passed which
+    have empty arrays as peaks. Functions shouldn't break in those cases."""
+    mz = numpy.array([], dtype="float")
+    intensities = numpy.array([], dtype="float")
+    spectrum_in = Spectrum(mz=mz, intensities=intensities,
+                           metadata={"testfield": "TEST"})
```
Comment (jspaaks) [inline tests/test_select_by_relative_intensity.py]: Why does this unit test have metadata? The point of the test is ``mz`` and ``intensities`` being empty arrays, right?

**Review by jspaaks — CHANGES_REQUESTED** [review]

**Comment by jspaaks** [comment]
BTW be aware that the changes in #150 will affect basically everything we had up to this point, so probably good to take a look at that before any other work.

File: tests/test_select_by_relative_intensity.py (hunk 2 above)
Comment (florian-huber) [reply tests/test_select_by_relative_intensity.py]: True! I removed the metadata from the test.

File: tests/test_normalize_intensities.py (hunk 1 above)
Comment (florian-huber) [reply tests/test_normalize_intensities.py]: True! I removed the metadata from the test.

**Review by jspaaks — APPROVED** [review]

**Comment by florian-huber** [comment]
@jspaaks Do you want to orchestrate the merging of the different PRs? 
#155 and then #156 will after merging probably imply some changes according to #150.
Let me know if I should do some of it.

**Comment by jspaaks** [comment]
I think I'm right in saying that
- #155 can now be merged into master
- this should make #156 mergable (I didn't check btw)
- #158 can be merged

If you agree, then feel free to go ahead and merge them (also delete any of your branches after they have been merged)

And then I will merge #150 sometime next week, but I wanted to give the you and the others opportunity to comment, even though @fdiblen kindly approved the PR already.

**Comment by florian-huber** [comment]
Thanks for reviewing Juriaan!

File: matchms/similarity/PrecursormzMatch.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+import numba
+import numpy
+from matchms.typing import SpectrumType
+from .BaseSimilarity import BaseSimilarity
+
+
+class PrecursormzMatch(BaseSimilarity):
+    """Return True if spectrums match in precursor m/z (within tolerance), and False otherwise.
+
+    Example to calculate scores between 2 spectrums and iterate over the scores
```
Comment (sverhoeven) [inline matchms/similarity/PrecursormzMatch.py]: I see 4 spectrums or 2 pairs of spectrums 
```suggestion
    Example to calculate scores between 2 pairs of spectrums and iterate over the scores
```

**Pull request merged**